app/interactors/file_attachments/preview_interactor.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class FileAttachments::PreviewInteractor
+  include Interactor
+
+  delegate :params,
+           :edition,
+           :attachment_revision,
+           :asset,
+           to: :context
+
+  def call
+    find_and_lock_edition
+    find_attachment
+    check_uploaded
+    check_available
+  end
+
+private
+
+  def find_and_lock_edition
+    context.edition = Edition.lock.find_current(document: params[:document])
+  end
+
+  def find_attachment
+    context.attachment_revision = edition.file_attachment_revisions
+      .find_by!(file_attachment_id: params[:file_attachment_id])
+  end
+
+  def check_uploaded
+    attachment_revision.ensure_assets
+    context.asset = attachment_revision.asset("file")
+
+    if asset.absent?
```

Comment:
As we just discussed in person, is it better to upload the attachment here which would probably mean the user wouldn't be able to access the attachment immediately meaning we would create a page to handle to the pending status?

Or to upload the attachment to asset manager when the file is originally uploaded which would mitigate the preview issue mentioned above. A possible downside I can see with this approach is if an asset was uploaded and deleted before the draft was saved then multiple requests would be sent to asset manager (however this scenario seems unlikely).

Sorry for repeating what we discussed but I thought i should put it in writing here too. :)

Reply:
@1pretz1 as discussed in person: this is something we could provide in feature if we see people are getting the 'pending' page a lot, but we would still have check_uploaded here as a decoupled fallback :-). We should do this again ;-).